src/common.rs

```diff
@@ -286,3 +286,78 @@ macro_rules! impl_step {
 }
 
 pub(crate) use impl_step;
+
+/// Implements support for the `num-traits` crate, if the feature is enabled.
+macro_rules! impl_num_traits {
+    ($type:ident, $primitive_8:ty, |$result:ident| $limit_result:expr) => {
+        #[cfg(feature = "num-traits")]
+        impl<T, const BITS: usize> num_traits::WrappingAdd for $type<T, BITS>
+        where
+            Self: Integer,
+            T: PartialEq
+                + Eq
+                + Copy
+                + Add<T, Output = T>
+                + Sub<T, Output = T>
+                + BitAnd<T, Output = T>
+                + Not<Output = T>
+                + Shr<usize, Output = T>
+                + Shl<usize, Output = T>
+                + From<$primitive_8>,
+            core::num::Wrapping<T>: Add<core::num::Wrapping<T>, Output = core::num::Wrapping<T>>,
+        {
+            #[inline]
+            fn wrapping_add(&self, rhs: &Self) -> Self {
+                let $result =
+                    (core::num::Wrapping(self.value()) + core::num::Wrapping(rhs.value())).0;
```

Comment:
it's been a while since I built this. Would it be sufficient/easier/more consistent to just use `self.value().wrapping_add(rhs.value())` instead? We don't use Wrapping elsewhere in this file

Reply:
By the way feel free to say no here. Don't want to put more work on you, just wanted to hear your opinion. Also happy to do this in a follow-up

Reply:
> it's been a while since I built this. Would it be sufficient/easier/more consistent to just use self.value().wrapping_add(rhs.value()) instead? We don't use Wrapping elsewhere in this file

We sadly can't. Because this implementation is generic over the underlying type the compiler doesn't know that `Self::wrapping_add()` is available, which we only implement for specific `T`s (for const-friendlyness).